inception-project/inception
# #2598 UI for the corpus statistics

Was: In this PR the UI for the corpus statistics will be build. It will be added to the annotation sidebar below search.


**What's in the PR**
- This PR adds a statistics sidebar to inception. It is located under Annotation directly below the search sidebar. 
- The statistics sidebar consists of two dropdowns where you can choose the granularity and the metric. 
- Next to the dropdowns there is a question mark which links to the User Guide. I added a detailed description of the sidebar in the asciidoc but it is not included yet in the official User Guide.
- A "hide empty layers"-checkbox 
- With the calculate button the calculation starts.
- The results of the calculation are shown in a table below the calculate button. It consists of 2 columns. 
- A format dropdown and an export button below the table to export the results in the desired format.

**How to test manually**
* For a detailed description, go to the User Guide -> Annotation -> Statistics (if it is updated already) or see the images below.
* Navigate to Annotation and click the clipboard icon to reach the statistics sidebar.
* Choose a granularity. per Document: the inputs for the metric are the raw counts per Document. per Sentence: before the raw counts are plugged into the metric, each of them is divided by the number of sentences in its respective documents.
* Choose a metric. They should be self-explanatory. 
* Choose whether to omit layer/features which do not have an annotation.
* Click on Calculate. This may take a while depending on project size.
* The results for your chosen metric are displayed in the table.
* If you want to change your choices of metric/granularity you have to click calculate again.
* Below the table you can export the whole table (also the bigger part which is not shown in the UI). For this, choose a format from the dropdown.

![grafik](https://user-images.githubusercontent.com/61779598/137896371-3869eac7-98e7-4ada-b1c5-3f3c4bd70b5c.png)
![grafik](https://user-images.githubusercontent.com/61779598/137896422-7c06c551-2de0-4a61-bc9b-eaa7c8da81b3.png)


**Automatic testing**

**Documentation**
